For legacy properly create cnt dir

michelmno opened the pull request:
fix issue #2142

Use Case: gwd started with -bd specifying a directory that is not current working dir (cwd)

Without this commit, each request reports warning in gwd.log "WARNING Failure when creating cnt"

git bisect identified cid ca2ebd7a4 as the first bad commit 
```
Author: Henri83 <henri.gouraud@laposte.net>
Date:   Sat Mar 1 15:33:06 2025 +0100
    dont create init files (cache, etc, cnt, ...) if base.gwb does not exist
```

In same commit remove lines creating etc in cwd,
that badly duplicate lines using etc_d.

michelmno committed:
For legacy properly create cnt

fix issue #2142

Use Case: gwd started with -bd specifying a directory that is not
current working dir (cwd)

Without this commit, each request reports warning in gwd.log
"WARNING Failure when creating cnt"

git bisect identified cid ca2ebd7a4 as the first bad commit
===
Author: Henri83 <henri.gouraud@laposte.net>
Date:   Sat Mar 1 15:33:06 2025 +0100
    dont create init files (cache, etc, cnt, ...) if base.gwb does not exist
===

limit code change to cnt,
potential same problem for etc and lang.

Signed-off-by: Michel Normand <michel_mno@laposte.net>

merge

hgouraud commented:
la modification proposée va dans le bon sens, mais mérite 5 min de réflexion.
Elle supprime la création de sous dossiers lang/lg! 
Il me semble que ces sous dossiers ne sont pas effectivement utilisés, et qu'en cas de besoin, ils pourraient être remplacés par la recherche de `lang/fichier-lg.txt` au lieu de `lang/lg/fichier.txt`. C'est déjà la solution adoptée dans GWPARAM.output_error
A noter que le problème se pose aussi pour les fichiers .txt dans le dossier src!

Plus généralement, il me semble que les dossiers etc, lang et src étant "read-only", il n'est pas nécessaire de les créer lors du démarrage de gwd! Le seul dossier en écriture est le dossier cnt.

A suivre

hgouraud committed:
fix warning 32 (GWPARAM) and 27 (gwdb-legacy/database)

michelmno commented:
mon patch est trés probablement trop étendu,  je ferais une mise à jour quand  je pourrais builder la branche master (pour le moment plante tel que remonté dans l'issue 2146)

En fait je ne m'intéresse que au dossier cnt, sachant que le code avant ce PR essaie de le créer dans le dossier courant au lieu du dossier bases dir, d'où le message de Warning que je veux faire disparaitre quand j'utilise la branche master !

michelmno committed:
fix warning GWPARAM oc not used in line 19

This is to complete previous correction of cid a1784b100
(hgouraud/fix-unused) fix warning 32 (GWPARAM) and 27 (gwdb-legacy/database)

michelmno commented:
> mon patch est trés probablement trop étendu, je ferais une mise à jour quand je pourrais builder la branche master (pour le moment plante tel que remonté dans l'issue 2146)
> 
> En fait je ne m'intéresse que au dossier cnt, sachant que le code avant ce PR essaie de le créer dans le dossier courant au lieu du dossier bases dir, d'où le message de Warning que je veux faire disparaitre quand j'utilise la branche master !

PR mis à jour en limitant le changement au dossier cnt ! (cid e7dc4dc
par contre  un commit venant du PR #2147 et une correction similaire, tous deux en attente de merge.

michelmno commented:
do not cover all conditions, so close the PR,  the issue is still valid
